src/custom-apps/components/WebhookEvents/WebhookEvents.tsx

```diff
@@ -180,59 +187,52 @@ export default WebhookEvents;
-  ATTRIBUTE: ["CREATED", "UPDATED", "DELETED"],
-  CATEGORY: ["CREATED", "UPDATED", "DELETED"],
-  CHANNEL: ["CREATED", "UPDATED", "DELETED"],
-  GIFT_CARD: ["CREATED", "UPDATED", "DELETED", "STATUS_CHANGED"],
-
-  CHECKOUT: ["CREATED", "UPDATED", "DELETED"],
-  COLLECTION: ["CREATED", "UPDATED", "DELETED"],
-  CUSTOMER: ["CREATED", "UPDATED", "DELETED"],
-  FULFILLMENT: ["CREATED"],
-  INVOICE: ["DELETED", "REQUESTED", "SENT"],
-  MENU: ["CREATED", "UPDATED", "DELETED"],
-  ORDER: [
-    "CANCELLED",
-    "CONFIRMED",
-    "CREATED",
-    "FULFILLED",
-    "FULLY_PAID",
-    "UPDATED",
-  ],
-  PAGE: ["CREATED", "UPDATED", "DELETED"],
-  PRODUCT: ["CREATED", "UPDATED", "DELETED"],
-  PRODUCT_VARIANT: ["CREATED", "UPDATED", "DELETED"],
-  SALE: ["CREATED", "UPDATED", "DELETED", "TOGGLE"],
-  SHIPPING_PRICE: ["CREATED", "UPDATED", "DELETED"],
-  SHIPPING_ZONE: ["CREATED", "UPDATED", "DELETED"],
-  STAFF: ["CREATED", "UPDATED", "DELETED"],
-  TRANSLATION: ["ACTION_REQUEST", "CREATED", "UPDATED"],
-  VOUCHER: ["CREATED", "UPDATED", "DELETED"],
-  WAREHOUSE: ["CREATED", "UPDATED", "DELETED"],
-};
+const getWebhookTypes = (webhookEvents: string[]) => {
+  const webhookTypes: Record<string, Actions> = {};
+  const multiWords = ["DRAFT_ORDER", "GIFT_CARD", "ANY_EVENTS"];
+
+  const setObject = (
+    keywords: string[],
+    object: string,
+    objectLength: number,
+  ) => {
+    const events = webhookTypes[object] || [];
+    const event = keywords.slice(objectLength).join("_");
+    events.push(!!event.length ? event : object);
+    webhookTypes[object] = events;
+  };
+
+  webhookEvents.map(key => {
+    const keywords = key.split("_");
+
+    // handle 2 word exceptions
+    if (multiWords.includes(keywords.slice(0, 2).join("_"))) {
+      setObject(keywords, keywords.slice(0, 2).join("_"), 2);
+      return;
+    }
+
+    setObject(keywords, keywords[0], 1);
+  });
 
-const SyncWebhookTypes: Record<string, Actions> = {
-  PAYMENT: [
-    "AUTHORIZE",
-    "CAPTURE",
-    "CONFIRM",
-    "LIST_GATEWAYS",
-    "PROCESS",
-    "REFUND",
-    "VOID",
-  ],
-  CHECKOUT: ["CALCULATE_TAXES", "FILTER_SHIPPING_METHODS"],
-  ORDER: ["CALCULATE_TAXES", "FILTER_SHIPPING_METHODS"],
-  SHIPPING: ["LIST_METHODS_FOR_CHECKOUT"],
+  return webhookTypes;
 };
 
+export const AsyncWebhookTypes: Record<string, Actions> = getWebhookTypes(
+  Object.keys(WebhookEventTypeAsyncEnum),
+);
+
+const SyncWebhookTypes: Record<string, Actions> = getWebhookTypes(
+  Object.keys(WebhookEventTypeSyncEnum),
+);
+
 const EventTypes = {
   async: AsyncWebhookTypes,
   sync: SyncWebhookTypes,
 };
 
-const getEventName = (object: string, event: string) =>
-  [object, event].join("_").toUpperCase() as WebhookEventTypeSyncEnum;
+const getEventName = (object: string, event: string) => {
+  if (object === event) {
+    return object.toUpperCase() as WebhookEventTypeSyncEnum;
+  }
+  return [object, event].join("_").toUpperCase() as WebhookEventTypeSyncEnum;
+};
```

Comment:
suggestion: move this code to separate file

Reply:
done